ndcube/asdf/converters/extracoords_converter.py

```diff
@@ -0,0 +1,29 @@
+from asdf.extension import Converter
+
+
+class ExtraCoordsConverter(Converter):
+    tags = ["tag:sunpy.org:ndcube/extra_coords/extra_coords/ExtraCoords-*"]
+    types = ["ndcube.extra_coords.extra_coords.ExtraCoords"]
+
+    def from_yaml_tree(self, node, tag, ctx):
+        from ndcube.extra_coords.extra_coords import ExtraCoords
+        extra_coords = ExtraCoords()
+        extra_coords._wcs = node.get("wcs")
```

Comment:
I can't remember if wcs is dynamic if we have lookup tables, so we wouldn't need to serialise it in that case. I can check later.

Reply:
ExtraCoords can be initialized by a `~astropy.wcs.wcsapi.BaseLowLevelWCS` object and a `mapping`. Therfore, I included `wcs` and `mapping` as a node key-value pair and when we initialize ExtraCoords with the `lookup_tables`, the `wcs` is indeed dynamic, and we don't need to serialize it. The structure is such that only when we initialize it with a `wcs` object and a `mapping`, it will get serialized. Did I understand it correctly, or am I missing something?

Reply:
You are pretty much right, there are two internal representations possible with ExtraCoords, first you have the case where someone makes an EC object like this:
```python
ec = ExtraCoords()
ec.wcs = my_wcs
```
the other is when you incrementally build up a EC object from tables:
```python
ec = ExtraCoords()
ec.add("coord1", ...)
```

We need to serialise these two cases differently. You can see from the implementation of `ExtraCoords.wcs`: https://github.com/sunpy/ndcube/blob/main/ndcube/extra_coords/extra_coords.py#L287 how this works. So I think we need the same `if ec._wcs` and `if ec._lookup_tables` logic in here as well.

Reply:
Looking below I see this is how you have it anyway! Looks good to me.